source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -77,8 +79,14 @@ Config::SharedConfig::SharedConfig(
     idle_timeout_ = std::chrono::hours(1);
   }
   if (config.has_tunneling_config()) {
-    tunneling_config_helper_ =
-        std::make_unique<TunnelingConfigHelperImpl>(config.tunneling_config());
+    envoy::config::core::v3::SubstitutionFormatString substitution_format_config;
```

Comment:
I think this code could be made part of `TunnelingConfigHelperImpl`. E.g. add a function there, and initialize const formatter field with a function call.

Reply:
Good idea, I did it.